plutus-tx/src/PlutusTx/Builtins.hs

```diff
@@ -282,6 +285,10 @@ emptyString = BI.emptyString
 equalsString :: BuiltinString -> BuiltinString -> Bool
 equalsString x y = fromBuiltin (BI.equalsString x y)
 
+{-# INLINABLE showConstant #-}
+showConstant :: GHC.Show.Show a => a -> BuiltinString
```

Comment:
This can't be right. These functions are supposed to be usable in on-chain code, but this will require a `GHC.Show` typeclass instance in the on-chain code? I don't think we can expose a function like this from here.

Reply:
Yeah, I think regardless of whether we do a fully polymorphic builtin, we probably want to expose it like that.